packages/jupyter-ai/jupyter_ai/handlers.py

```diff
@@ -115,12 +115,23 @@ def get_chat_user(self) -> ChatUser:
         )
 
         if collaborative:
-            return ChatUser(**asdict(self.current_user))
+            names = self.current_user.name.split(" ", maxsplit=2)
+            initials = "".join(
+                [(name.capitalize()[0] if len(name) > 0 else "") for name in names]
+            )
+            chat_user_kwargs = {
+                # set in case IdentityProvider doesn't return initials, e.g.
+                # JupyterHub (#302)
+                "initials": initials,
```

Comment:
In the case where initials is set to None by the identity provider, this doesn't work. This line should be *below* the following line.